plugins/language_json.lua

```diff
@@ -0,0 +1,28 @@
+-- mod-version:3 priority:90
+
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "JSON",
+  files = { "%.json$" },
+  comment = nil,
+  patterns = {
+
+    { pattern = '\"[a-zA-Z0-9_]*\":', type = "keyword" }, -- key (string)
+    { pattern = '[0-9_]*:', type = "keyword" }, -- key (int)
+    { pattern = { '"', '"', "\\"}, type = "string" }, -- value
+
+    -- numbers
+    { pattern = "0x[%da-fA-F]+", type = "number" },
+    { pattern = "-?%d+[%d%.eE]*",  type = "number" },
+    { pattern = "-?%.?%d+", type = "number" },
+
+    -- literals
+    { pattern = "NaN", type = "literal" },
+    { pattern = "Infinity", type = "literal" },
```

Comment:
I wonder if those should be removed, as they're invalid JSON.

Reply:
I only added them because I've seen it in real world examples, but yeah as they're invalid JSON I think it's best to strip them, or let the user re-add them incase it's neccesary